### PR TITLE
Refactor repl mode toggler test

### PR DESCRIPTION
**Pull request by aadcg**
# Description

This is a follow up to #2827 with respect to testing. Sorry that I didn't notice it in the PR!

# Discussion

@aartaka, it seems to me that you didn't take into consideration the comments relative to the repl mode toggler test that was not active.

I've made a simple refactor that brings that test back (we have a basic mode toggling tests for all modes) and updated the outdated comments.

# Checklist:
Everything in this checklist is required for each PR.  Please do not approve a PR that does not have all of these items.

- [x] I have pulled from master before submitting this PR
- [x] There are no merge conflicts.
- [x] I've added the new dependencies as:
  - [x] ASDF dependencies,
  - [x] Git submodules,
    ```sh
	cd /path/to/nyxt/checkout
    git submodule add https://gitlab.common-lisp.net/nyxt/py-configparser _build/py-configparser
    ```
  - [x] and Guix dependencies.
- [x] My code follows the style guidelines for Common Lisp code. See:
  - [Norvig & Pitman's Tutorial on Good Lisp Programming Style (PDF)](https://www.cs.umd.edu/~nau/cmsc421/norvig-lisp-style.pdf)
  - [Google Common Lisp Style Guide](https://google.github.io/styleguide/lispguide.xml)
- [x] I have performed a self-review of my own code.
- [x] My code has been reviewed by at least one peer.  (The peer review to approve a PR counts.  The reviewer must download and test the code.)
- [x] Documentation:
  - [x] All my code has docstrings and `:documentation`s written in the aforementioned style.  (It's OK to skip the docstring for really trivial parts.)
  - [x] I have updated the existing documentation to match my changes.
  - [x] I have commented my code in hard-to-understand areas.
  - [ ] I have updated the `changelog.lisp` with my changes if it's anything user-facing (new features, important bug fix, compatibility breakage).
  - [x] I have added a `migration.lisp` entry for all compatibility-breaking changes.
  - [x] (If this changes something about the features showcased on Nyxt website) I have these changes described in the new/existing article at Nyxt website or will notify one of maintainters to do so.
- [x] Compilation and tests:
  - [x] My changes generate no new warnings.
  - [x] I have added tests that prove my fix is effective or that my feature works.  (If possible.)
  - [x] New and existing unit tests pass locally with my changes.

### CODE REVIEW

File: tests/offline/mode/repl.lisp [hunk 1]
```diff
@@ -3,12 +3,16 @@
...
+  (let ((buffer (make-instance 'modable-buffer)))
+    (with-current-buffer buffer
+      (assert-true (enable-modes* 'nyxt/repl-mode:repl-mode buffer))
+      (assert-true (disable-modes* 'nyxt/repl-mode:repl-mode buffer)))))
+
 (define-test evaluate-repl-mode ()
   ;; Needed because most REPL commands reload the buffer.
   (let ((buffer (make-instance 'web-buffer)))
     (with-current-buffer buffer
-      (assert-true (enable-modes* 'nyxt/repl-mode:repl-mode buffer))
```
Comment (aartaka) [inline tests/offline/mode/repl.lisp:11]: Note that this line was required to test the functionality of `repl-mode`. The mode has to be enables to test what it does, after all (:

**Review by aartaka — APPROVED** [review]

File: tests/offline/mode/repl.lisp (hunk 1 above)
Comment (aadcg) [reply tests/offline/mode/repl.lisp:11]: Indeed, my mistake! I've pushed a fix.

**Comment by aartaka** [comment]
All OK, thanks for the fix!

**Comment by aadcg** [comment]
Thanks @aartaka. Pushed as individual commits, as I didn't see the need to push it as a branch (trivial commits and all atomic).

**Pull request merged**